lib/ansible/module_utils/network/eos/config/interfaces/interfaces.py

```diff
@@ -0,0 +1,250 @@
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = _param_list_to_dict(want)
+        have = _param_list_to_dict(have)
+        state = self._module.params['state']
+        if state == 'overridden':
+            commands = self._state_overridden(want, have)
+        elif state == 'deleted':
+            commands = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands = self._state_merged(want, have)
+        elif state == 'replaced':
+            commands = self._state_replaced(want, have)
+        return commands
+
+    @staticmethod
+    def _state_replaced(want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = _compute_commands(want, have, replace=True, remove=True)
+
+        replace = commands['replace']
+        remove = commands['remove']
+
+        commands_by_interface = replace
+        for interface, commands in remove.items():
+            commands_by_interface[interface] = replace.get(interface, []) + commands
+
+        return _flatten_commands(commands_by_interface)
+
+    @staticmethod
+    def _state_overridden(want, have):
+        """ The command generator when state is overridden
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        # Add empty desired state for unspecified interfaces
+        for key in have:
+            if key not in want:
+                want[key] = {}
+
+        # Otherwise it's the same as replaced
+        return Interfaces._state_replaced(want, have)
+
+    @staticmethod
+    def _state_merged(want, have):
+        """ The command generator when state is merged
+
+        :rtype: A list
+        :returns: the commands necessary to merge the provided into
+                  the current configuration
+        """
+        commands = _compute_commands(want, have, replace=True)
+        return _flatten_commands(commands['replace'])
+
+    @staticmethod
+    def _state_deleted(want, have):
+        """ The command generator when state is deleted
+
+        :rtype: A list
+        :returns: the commands necessary to remove the current configuration
+                  of the provided objects
+        """
+        commands = _compute_commands(want, have, remove=True)
+        return _flatten_commands(commands['remove'])
+
+
+def _param_list_to_dict(param_list):
```

Comment:
If this is a common method probably can be moved to `utils` with an additional argument to handle `name` key?

Reply:
yeah, it'd be good to add that to utils so it can be reused in the future.